src/modules/FileActionsMenu/FileActionsMenu/dllmain.cpp

```diff
@@ -0,0 +1,426 @@
+            return true; // Might as well assume it's possible it's an explorer window.
+        }
+
+        // Enumerate all Shell Windows to compare the window handle against.
+        IUnknownPtr spEnum{};
+        result = spShellWindows->_NewEnum(&spEnum);
+        if (result != S_OK || spEnum == nullptr)
+        {
+            Logger::warn(L"Failed to list explorer Windows. {}", GetErrorString(result));
+            return true; // Might as well assume it's possible it's an explorer window.
+        }
+
+        IEnumVARIANTPtr spEnumVariant{};
+        result = spEnum.QueryInterface(__uuidof(spEnumVariant), &spEnumVariant);
+        if (result != S_OK || spEnumVariant == nullptr)
+        {
+            Logger::warn(L"Failed to enum explorer Windows. {}", GetErrorString(result));
+            spEnum->Release();
+            return true; // Might as well assume it's possible it's an explorer window.
+        }
+
+        variant_t variantElement{};
+        while (spEnumVariant->Next(1, &variantElement, NULL) == S_OK)
+        {
+            IWebBrowserApp* spWebBrowserApp;
+            result = variantElement.pdispVal->QueryInterface(IID_IWebBrowserApp, reinterpret_cast<void**>(&spWebBrowserApp));
+            if (result == S_OK)
+            {
+                HWND hwnd;
+                result = spWebBrowserApp->get_HWND(reinterpret_cast<SHANDLE_PTR*>(&hwnd));
+                if (result == S_OK)
+                {
+                    if (hwnd == windowHandle)
+                    {
+                        VariantClear(&variantElement);
+                        spWebBrowserApp->Release();
+                        spEnumVariant->Release();
+                        spEnum->Release();
+                        return true;
+                    }
+                }
+                spWebBrowserApp->Release();
+            }
+            VariantClear(&variantElement);
+        }
+
+        spEnumVariant->Release();
+        spEnum->Release();
+        return false;
+    }
+
+    bool is_FileActionsMenu_or_explorer_or_desktop_window_focused()
+    {
+        HWND foregroundWindowHandle = GetForegroundWindow();
+        if (foregroundWindowHandle == NULL)
+        {
+            return false;
+        }
+
+        DWORD pid{};
+        if (GetWindowThreadProcessId(foregroundWindowHandle, &pid)!=0)
+        {
+            // If the foreground window is the FileActionsMenu window, send activation signal.
+            if (m_processPid != 0 && pid == m_processPid)
+            {
+                return true;
+            }
+        }
+
+        if (is_desktop_window(foregroundWindowHandle))
+        {
+            return true;
+        }
+
+        return is_explorer_window(foregroundWindowHandle);
+    }
+
+    void launch_process()
+    {
+        Logger::trace(L"Starting FileActionsMenu.UI process");
+
+        std::wstring executable_args = L"";
+
+        Logger::trace("Starting FileActionsMenu non elevated from elevated process");
```

Comment:
no need for this trace. we don't know if process is elevated. trace from few lines above is enough